operators/pkg/controller/elasticsearch/version/version7/initial_master_nodes.go

```diff
@@ -53,3 +69,26 @@ func ClusterInitialMasterNodesEnforcer(
 
 	return &performableChanges, nil
 }
+
+// isBootstrapped checks if the cluster has already been bootstrapped
+func isBootstrapped(cluster v1alpha1.Elasticsearch, clusterState observer.State, c k8s.Client) (bool, error) {
+	_, ok := cluster.Annotations[ClusterUUIDAnnotationName]
+	if ok {
+		// existence of the annotation shows that the cluster has been bootstrapped
+		return true, nil
+	}
+
+	// no annotation, let see if the cluster has been bootstrapped by looking at it's UUID
+	if clusterState.ClusterState != nil && len(clusterState.ClusterState.ClusterUUID) > 0 {
```

Comment:
I don't like that `isBootstrapped` has the side effect of actually storing the cluster UUID on the ES resource.

Reply:
Since it is rather small I have inlined the function and make it explicit in the `ClusterInitialMasterNodesEnforcer` documentation.